modules/messages/src/inbound_lane.rs

```diff
@@ -114,25 +115,29 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 		}
 
 		let push_new = match data.relayers.back_mut() {
-			Some((_, nonce_high, last_relayer)) if last_relayer == &relayer => {
+			Some((_, nonce_high, last_relayer)) if last_relayer == relayer_at_brdiged_chain => {
 				*nonce_high = nonce;
 				false
 			}
 			_ => true,
 		};
 		if push_new {
-			data.relayers.push_back((nonce, nonce, relayer));
+			data.relayers
+				.push_back((nonce, nonce, (*relayer_at_brdiged_chain).clone()));
 		}
 
 		self.storage.set_data(data);
 
-		P::dispatch(DispatchMessage {
-			key: MessageKey {
-				lane_id: self.storage.id(),
-				nonce,
+		P::dispatch(
+			relayer_at_this_chain,
```

Comment:
We should probably add a note somewhere, that relayers should make sure that their derived accounts are above `ED` cause otherwise they might not be able to collect any fees.

Reply:
Yes, but here (at target chain) this problem is not super-important - the relayer account is the account that submits delivery tx. So it is guaranteed to have some funds (at least to pay for the tx). Te edge case is when relayer account balance drops under ED as a result of the delivery tx and fee is less than ED (this unlikely to happen with adequate relayer, because he'll be unable to submit any more transactions).

Reply:
Ah good point, I forgot about this.